### PR TITLE
fix(transformData): default data is an object when not provided

### PR DESCRIPTION
**Pull request by vvo**
This fixes the nasty regression bug we have seen and reimplement the
transformData fix

### CODE REVIEW

File: src/components/Template.js [hunk 1]
```diff
@@ -137,4 +137,17 @@ function transformHelpersToHogan(helpers, compileOptions, data) {
...
 }
 
-export default Template;
+// Resolve transformData before Template, so transformData is always called
+// even if the data is the same. Allowing you to dynamically inject conditions in
+// transformData that will force re-rendering
+const withTransformData =
+  TemplateToWrap =>
+    props => {
+      const data = props.data === undefined ? {} : props.data; // eslint-disable-line react/prop-types
```
Comment (bobylito) [inline src/components/Template.js]: Well it fixes the problem and is similar to what we were doing already.

Comment (vvo) [inline src/components/Template.js]: I think it happens in currentRefinedValues where you can easily `create` items via attributes and those might not yet be refined.

**Comment by vvo** [comment]
Added a test to ensure we default to {}

**Comment by bobylito** [comment]
The test looks fine :)

**Pull request merged**